org.adempiere.base/src/org/compiere/model/MTree_Base.java

```diff
@@ -406,12 +406,12 @@ protected boolean beforeSave (boolean newRecord)
 		
 		String tableName = getSourceTableName(true);
 		MTable table = MTable.get(getCtx(), tableName);
-		if (table.getColumnIndex("IsSummary") < 0) {
+		if (! table.columnExists("IsSummary")) {
 			// IsSummary is mandatory column to have a tree
 			log.saveError("Error", "IsSummary column required for tree tables"); 
 			return false;
 		}
-		if (table.getColumnIndex("Value") < 0) {
+		if (! table.columnExists("Value")) {

```

Comment:
The Value column here needs to exist in the database. Complete if clause:
if (! table.columnExists("Value") || table.getColumn("Value").isVirtual()) ...

Reply:
Thanks @Soundbytes - you're right - I tracked the 10 cases where that's used in core and all of them lead to a query using that column.

So, I decided is safer to add the method columnExistsNonVirtual - whoever wants to check the existence of virtual column can still use the getColumnIndex approach.

Reply:
@CarlosRuiz-globalqss Why not leave columnExists() as a check wether the column exists in the dictionary? After all all the code examples using getColumnIndex() for the purpose will be gone after this patch is merged.

Reply:
What I didn't like leaving that method is that it can create again the confusion.  When a developer is looking for such method it will find first columnExists and probably will just use that one without looking if there is another.

Reply:
I see. Then maybe rename it to columnExistsInDictionary. I still find the getColumnIndex() approach error prone and now with no code examples available even more so. columnExistsNonVirtual could then be renamed to columnExistsInDB. Then it is alphabetically listed before the other method.

Reply:
OK - done